lib/modules/dosomething/dosomething_helpers/dosomething_helpers.share.inc

```diff
@@ -26,7 +26,7 @@ function dosomething_helpers_share_redirect() {
  *
  * @return string   The facebook intent url.
  */
-function dosomething_helpers_get_facebook_intent($link, $type = NULL, $custom_options = NULL) {
+function dosomething_helpers_get_facebook_intent($type = NULL, $link, $custom_options = NULL) {
```

Comment:
let's switch type and link around to the way it was originally
you want to leave arguments with defaults at the end, so you can leave arguments with defaults out of a func call